cve_bin_tool/VersionSignature.py

```diff
@@ -50,26 +50,35 @@ def get_mapping_data(self):
         """
         self.cursor.execute(
             'CREATE TABLE IF NOT EXISTS {}(version TEXT , sourceId TEXT PRIMARY KEY)'.format(self.table_name))
-        # fetch first entry
-        entry = self.cursor.execute('SELECT * FROM {}'.format(self.table_name)).fetchone()
+
+        self.cursor.execute(
+            'CREATE TABLE IF NOT EXISTS {} (date TEXT PRIMARY KEY)'.format('last_update_' + self.table_name)
+        )
+
         update_required = False
+
+        entry = self.cursor.execute('SELECT * FROM {}'.format('last_update_'+self.table_name))
+
         if entry is not None:
-            # Check if update is required
-            # Updates if the difference between current time and the time of latest release is greater than duration
-            cur_time = datetime.datetime.now().replace(microsecond=0)
-            time_fields = entry[1].split()
-            year_data = time_fields[0].split('-')
-            time_data = time_fields[1].split(':')
-            yy, mm, dd = int(year_data[0]), int(year_data[1]), int(year_data[2])
-            H, M, S = int(time_data[0]), int(time_data[1]), int(time_data[2])
-            last_release = datetime.datetime(yy, mm, dd, H, M, S)
-            time_elapsed = str(cur_time - last_release)
-            if int(time_elapsed.split()[0]) >= self.duration:
+            # Updates if the difference between current time and the time of last update is greater than duration
+
+            # create a date object from the string of format 'yyyy-mm-dd'
+            update_data = entry.split('-')
+            last_update = date(int(update_data[0]), int(update_data[1]), int(update_data[2]))
+
```

Comment:
Using split might be the easier way to do it and be python2.7 & 3 compatible, but it's worth noting that there are built in parsers for this:

```python
import datetime
import pytz
import dateutil.parser
format = '%Y-%m-%d'
datestring = '2019-03-21'
d = dateutil.parser.parse(datestring)   # python 2.7
d = datetime.datetime.strptime(datestring, format) #3.2+
```

Reply:
`strptime` seems to work on python 2.7 as well